File: backend/entity/showdate.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type ShowDate struct {
	gorm.Model
	ConcertID uint `json:"concert_id"`
	Concert *Concert  `gorm:"foreignkey:ConcertID" json:"concert"`
	VenueID uint `json:"venue_id"`
	Venue *Venue `gorm:"foreignKey:VenueID" json:"venue"`
	ShowDate time.Time `json:"show_date"`
}

File: frontend/src/interfaces/showdate.ts
import type { AssignmentInterface } from "./assignment";
import type { ConcertInterface } from "./concert";
import type { VenueInterface } from "./venue";

export interface ShowDateInterface {
    ID?: number;
    concert_id?: number; 
    venue_id?: number; 
    show_date: string; 
    assignments: AssignmentInterface[];

    concert: ConcertInterface;
    venue: VenueInterface;
}
